eyeshade/controllers/referrals.js

```diff
@@ -74,29 +74,39 @@ v1.createReferrals = {
       const payload = request.payload
       const debug = braveHapi.debug(module, request)
       const referrals = runtime.database.get('referrals', debug)
-      let entries, matches, probi, query
+      const downloadIdsToBeConfirmed = []
+      let entries, existingDownloadIds, probi
 
       entries = await referrals.find({ transactionId: transactionId })
-      if (entries.length > 0) return reply(boom.badData('existing transaction-identifier: ' + transactionId))
+      if (entries.length > 0) return reply({})
```

Comment:
I wonder if we should avoid early return here. as it's possible (although unlikely) that some referrals were added to the collection and others were not

Reply:
Yes, you're right.  I don't think we care if there already is a referral with the same transaction id.  We should only care about download id.  I will remove this line.

In practice though I don't think this should be too much of a problem since txId is effectively a request id, since a new one is used every request.  So in the case you described, the referrals should just go through on the next request.

Reply:
Added a fix + test in 07c466d